src/generators/go/constrainer/EnumConstrainer.ts

```diff
@@ -22,44 +23,27 @@ export const DefaultEnumKeyConstraints: ModelEnumKeyConstraints = {
   NO_EMPTY_VALUE,
   NAMING_FORMATTER: FormatHelpers.toConstantCase,
   NO_RESERVED_KEYWORDS: (value: string) => {
-    return value;
+    return NO_RESERVED_KEYWORDS(value, isReservedGoKeyword); 
   }
 };
 
-export type EnumKeyContext = {
-  enumKey: string,
-  constrainedEnumModel: ConstrainedEnumModel,
-  enumModel: EnumModel
-}
-export type EnumConstraintType = (context: EnumKeyContext, constraints?: ModelEnumKeyConstraints) => string;
+export function defaultEnumKeyConstraints(customConstraints?: Partial<ModelEnumKeyConstraints>): EnumKeyConstraint {
+  const constraints = {...DefaultEnumKeyConstraints, ...customConstraints};
```

Comment:
```suggestion
export function defaultEnumKeyConstraints(customConstraints: Partial<ModelEnumKeyConstraints> = {}): EnumKeyConstraint {
  const constraints = {...DefaultEnumKeyConstraints, ...customConstraints};
```

You cannot destroy the `undefined/null`.

Reply:
The current setup works fine 🙂 

`defaultEnumKeyConstraints(undefined);` works as expected. 

I have added the following test:

```
    test('should be able to handle undefined', () => {
      const constrainFunction = defaultEnumKeyConstraints(undefined);
      const value = constrainFunction({enumModel, constrainedEnumModel, enumKey: 'TEST'});
      expect(value).toEqual('TEST');
    });
```

Reply:
🤣  I always think that undefined/null will throw error